optimum/intel/ipex/inference.py

```diff
@@ -20,7 +31,48 @@
     import intel_extension_for_pytorch as ipex
 
 
-class _ModelFallbackWrapper:
+def ordered_inputs(inputs, model):
+    ordered_inputs = {}
+    if hasattr(model, "forward"):
+        sig = inspect.signature(model.forward)
+    else:
+        sig = inspect.signature(model.call)
+
+    for param in sig.parameters:
+        param_regex = re.compile(rf"{param}(\.\d*)?")
+        to_insert = []
+        for name, dynamic_axes in inputs.items():
+            if re.match(param_regex, name):
+                to_insert.append((name, dynamic_axes))
+        # TODO: figure out a smart way of re-ordering potential nested structures.
+        to_insert = sorted(to_insert, key=lambda t: t[0])
+        for name, dynamic_axes in to_insert:
+            ordered_inputs[name] = dynamic_axes
+    return ordered_inputs

```

Comment:
Wouldn't this work :    
```suggestion
    return tuple(inputs[key] for key in sig.parameters if inputs.get(key, None) is not None)
```